aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadTask.java

```diff
@@ -233,11 +226,16 @@ private Boolean uploadMultipartAndWaitForCompletion() throws ExecutionException
 
         LOGGER.info("Completing the multi-part upload transfer for " + upload.id);
         try {
-            completeMultiPartUpload(upload.id, upload.bucketName, upload.key,
+            CompleteMultipartUploadResult result = completeMultiPartUpload(upload.id, upload.bucketName, upload.key,
                     upload.multipartId);
-            updater.updateProgress(upload.id, upload.bytesTotal, upload.bytesTotal, true);
-            updater.updateState(upload.id, TransferState.COMPLETED);
-            return true;
+            if (result != null) {
+                updater.updateProgress(upload.id, upload.bytesTotal, upload.bytesTotal, true);
+                updater.updateState(upload.id, TransferState.COMPLETED);
+                return true;
+            } else {
+                updater.updateState(upload.id, TransferState.FAILED);
+                return false;
+            }
         } catch (final AmazonClientException ace) {
             LOGGER.error("Failed to complete multipart: " + upload.id
                     + " due to " + ace.getMessage(), ace);
```

Comment:
Is there a missing call here to abort the multipart upload?

Reply:
Added a call to abort the transfer when complete multipart failed.